File: atividade_02/JavaScript/main.js
const path = require('path');
const MaquinaDeLerArquivos = require('./MaquinaDeLerArquivos.js');

function main() {
    
    const maquinaTopada = new MaquinaDeLerArquivos();
    maquinaTopada.debugMode = false;

    // Usar o separador de arquivo apropriado para o sistema operacional
    const scriptDir = __dirname;
    const arquivo = path.join(scriptDir, "arq-novo.txt");
    console.log(`Lendo arquivo: ${arquivo}`);
    maquinaTopada.lerArquivo(arquivo);
}

if (require.main === module) {
    main();
}

File: atividade_02/Python/main.py
from maquina_de_ler_arquivos import MaquinaDeLerArquivos
import os

def main():
    maquina_topada = MaquinaDeLerArquivos()
    maquina_topada.debug_mode = False

    # Usar o separador de arquivo apropriado para o sistema operacional
    script_dir = os.path.dirname(os.path.abspath(__file__))
    arquivo = os.path.join(script_dir, "arq-novo.txt")
    print(f"Lendo arquivo: {arquivo}")
    maquina_topada.ler_arquivo(arquivo)

if __name__ == "__main__":
    main()